### PR TITLE
[easy] Create functions to access columns as slices

### PR DESCRIPTION
**Pull request by querolita**
The reason for this PR is that when columns get reduced by ~800, there won't be any more vectors inside `KeccakColumns` for individual states, and we need some helpers to access them.

### CODE REVIEW

**Review by mrmr1993 — APPROVED** [review]

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
After looking over later PRs, it doesn't seem like we actually need to clone these, and it has a negative impact on the performance. Could you return slice references instead?

File: optimism/src/keccak/constraints.rs [hunk 1]
```diff
@@ -244,9 +245,10 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
             } // END chi
 
             // STEP iota: 4 constraints
-            for (q, c) in self.round_constants().iter().enumerate() {
+            for (q, c) in self.round_constants().to_vec().iter().enumerate() {
```
Comment (mrmr1993) [inline optimism/src/keccak/constraints.rs:248]: If you're going to clone anyway (via `to_vec`), you should probably just use `into_iter` and change `c.clone()` to `c` below.

File: optimism/src/keccak/constraints.rs [hunk 2]
```diff
@@ -73,22 +73,23 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
 
         // SPONGE CONSTRAINTS
         {
-            for z in self.sponge_zeros() {
+            for i in 0..SPONGE_ZEROS_LEN {
                 // Absorb phase pads with zeros the new state
-                self.constrain(self.is_absorb() * z.clone());
+                self.constrain(self.is_absorb() * self.sponge_zeros()[i].clone());
```
Comment (mrmr1993) [inline optimism/src/keccak/constraints.rs:78]: If this is going to be immediately cloned anyway, this makes sense to be a vector.

File: optimism/src/keccak/constraints.rs [hunk 3]
```diff
@@ -73,22 +73,23 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
...
             }
             for i in 0..QUARTERS * DIM * DIM {
                 // In first absorb, root state is all zeros
-                self.constrain(self.is_root() * self.old_state(i));
+                self.constrain(self.is_root() * self.old_state(i).clone());
```
Comment (mrmr1993) [inline optimism/src/keccak/constraints.rs:82]: Likewise for these.

**Review by mrmr1993 — COMMENTED** [review]
The conversion from `Vec` to `[_]` might have been a bit overzealous. It seems like it doesn't make sense for those values that are now cloned immediately.

**Review by mrmr1993 — COMMENTED** [review]
Approving to unblock, getting the slice/vec balance right is just a nit.

**Review by mrmr1993 — APPROVED** [review]

File: optimism/src/keccak/constraints.rs (hunk 1 above)
Comment (querolita) [reply optimism/src/keccak/constraints.rs:248]: Thanks for finding the nit, I will trace this issue here and address it asap.

https://github.com/o1-labs/proof-systems/issues/1678

**Pull request merged**